src/components/button-bar/button-bar.test.tsx

```diff
@@ -0,0 +1,192 @@
+import React from "react";
+import { render, screen } from "@testing-library/react";
+
+import Button from "../button";
+import Icon from "../icon";
+import IconButton from "../icon-button";
+import ButtonBar from "./button-bar.component";
+
+describe("When ButtonBar children are Button components", () => {
+  it("should render the Buttons as expected", () => {
+    render(
+      <ButtonBar>
+        <Button aria-label="button-in-bar">foo</Button>
+        <Button aria-label="button-in-bar">bar</Button>
+        <Button aria-label="button-in-bar">baz</Button>
+      </ButtonBar>
+    );
+
+    const buttonChildren = screen.getAllByRole("button", {
+      name: "button-in-bar",
+    });
+
+    expect(buttonChildren).toHaveLength(3);
+  });
+
+  it("should render the Buttons with icons as expected when the 'iconType' prop is set", () => {
+    render(
+      <ButtonBar>
+        <Button aria-label="button-in-bar" iconType="bin">
+          foo
+        </Button>
+        <Button aria-label="button-in-bar" iconType="bin">
+          bar
+        </Button>
+        <Button aria-label="button-in-bar" iconType="bin">
+          baz
+        </Button>
+      </ButtonBar>
+    );
+
+    const buttonChildren = screen.getAllByRole("button", {
+      name: "button-in-bar",
+    });
+    const icons = screen.getAllByTestId("icon");
+
+    expect(buttonChildren).toHaveLength(3);
+    expect(icons).toHaveLength(3);
+  });
+
+  it("should render the Buttons with 'size' prop set to 'medium'", () => {
+    render(
+      <ButtonBar>
+        <Button aria-label="button-in-bar">foo</Button>
+        <Button aria-label="button-in-bar">bar</Button>
+        <Button aria-label="button-in-bar">baz</Button>
+      </ButtonBar>
+    );
+
+    const buttonChildren = screen.getAllByRole("button", {
+      name: "button-in-bar",
+    });
+
+    buttonChildren.forEach((button) => {
+      expect(button).toHaveStyle("font-size: var(--fontSizes100)");
+      expect(button).toHaveStyle("min-height: 40px");
+    });
+  });
+
+  it("should render the Buttons with the 'iconPosition' prop set to 'before'", () => {
```

Comment:
**suggestion(non-blocking)**: how about we amend a few of test titles to make the intended behavior clearer? This way, if something breaks and a test fails, it will be easier for us to understand what the component is supposed to do.

For example:
```tsx
it("renders icon with right margin when iconPosition prop is 'before'");
it("does not render Buttons with full width when fullWidth prop is false");
```

I appreciate that most of the tests are checking trivial CSS rules though, so I'll leave this up to you. What are your thoughts?